core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -303,6 +309,33 @@ public static List<DataFile> newFiles(
     return newFiles;
   }
 
+  public static CloseableIterable<DataFile> newFilesBetween(
+      Long baseSnapshotId, long latestSnapshotId, Function<Long, Snapshot> lookup, FileIO io) {
```

Comment:
Nit: I know this is just using what the previous API did but I feel like better names would be `startSnapshotId` and `endSnapshotId` similar naming to some of the incremental structures in Iceberg.